### ISSUE TITLE
[Question] Azure OpenAI - Latest API version missing "image" message content for Assistant and promised preview features

### ISSUE THREAD

**Issue by hayescode** [Azure/azure-rest-api-specs#30090]
### API Spec link

https://github.com/Azure/azure-rest-api-specs/tree/main/specification/cognitiveservices/data-plane/AzureOpenAI

### API Spec version

2024-07-01-preview

## Message Content
When trying to add non-text messages we get an error message indicating only "text" is allowed. OpenAI has supported this for a couple months so it's surprising to see this missing still. Do we have an ETA on when this will be added?
```py
await client.beta.threads.messages.create(
    thread_id=thread.id,
    role="user",
    content=[{'text': 'what does this say?', 'type': 'text'}, {'image_url': {'url': 'data:image/png;base64,iVBORw0KG...', 'detail': 'auto'}, 'type': 'image_url'}]
)
```
```bash
openai.BadRequestError: Error code: 400 - {'error': {'message': "Invalid message content: only messages with 'type=text' are supported currently.", 'type': 'invalid_request_error', 'param': 'content', 'code': 'invalid_type'}}
```

## Promised Features
[This post](https://techcommunity.microsoft.com/t5/ai-azure-ai-services-blog/announcing-azure-openai-service-assistants-public-preview/ba-p/4143217) from Microsoft Build said Assistants would get `Bring Your Indexes to File Search` and `Browse` in July 2024, but yesterday's API release doesn't seem to include this. Is there an updated release timeline so we can properly plan for this?

**Comment by hao1300** [comment]
We recently ran into this issue as well? Any update on this?

**Comment by wilson101xx** [comment]
Same here any updates?

**Comment by hayescode** [comment]
@jpalvarezl @trrwilson Will this be fixed in #30909 ? I've tried `2024-08-01-preview` and `2024-09-01-preview` but I still get the same error.

**Comment by sorieux** [comment]
I’m also encountering this issue with 2024-09-01-preview, and it’s blocking progress on one of my projects. Are there any workarounds available to enable using images with an assistant?

**Comment by hayescode** [comment]
I'm guessing it comes from [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/OpenAI.Assistants/messages/models.tsp#L21-L24), which is using the 2024-02-15 version of Assistants. Azure OpenAI is falling far behind OpenAI in many cases it seems they can't maintain parity :(

**Comment by rrosztoczy** [comment]
I'm also waiting for this.

**Comment by pedrion** [comment]
Any updates ? I also have this issue @jpalvarezl  @trrwilson

**Comment by MMarciniak103** [comment]
Hey, I'm also experiencing the same issue. Is there any update?

**Comment by MatthiasKerat** [comment]
We are also waiting on this. Still only text prompt is available.

**Comment by MarkusvonStaden** [comment]
Also still waiting to get this fixed. This puts my project on hold.

**Comment by bahadiraraz** [comment]
same

**Comment by strongwinderic** [comment]
same

**Comment by hayescode** [comment]
@glecaros @jpalvarezl @trrwilson will the ability to add image input for Assistants using models like GPT4o reach parity with OpenAI and the Chat Completions API in #32256 ?

**Comment by zetomatoz** [comment]
I’m facing the same issue. It would be great if the Azure OpenAI Assistant API could align more quickly with the capabilities of the OpenAI API. At the very least, the Azure OpenAI documentation should explicitly mention that the messages functionality does not yet support images. Currently, the documentation is quite misleading as it seems to suggest otherwise ([source](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/assistant#assistants-components)).

As a workaround, you can use a vision-enabled chat to analyze the image and then pass its output to the assistant for further processing. While this works, it’s not an ideal solution.

**Comment by hayescode** [comment]
@YunsongB @johanste #34354 seems to have removed image inputs for Assistants API. This was working last month for GPT 4o.


```bash
BadRequestError: Error code: 400 - {'error': {'message': 'Invalid model: `gpt-4.1-2025-04-14` does not support image message content types.', 'type': 'invalid_request_error', 'param': 'model', 'code': 'invalid_type'}}
```

**Comment by harithzulfaizal** [comment]
> [@YunsongB](https://github.com/YunsongB) [@johanste](https://github.com/johanste) [#34354](https://github.com/Azure/azure-rest-api-specs/pull/34354) seems to have removed image inputs for Assistants API. This was working last month for GPT 4o.
> 
> BadRequestError: Error code: 400 - {'error': {'message': 'Invalid model: `gpt-4.1-2025-04-14` does not support image message content types.', 'type': 'invalid_request_error', 'param': 'model', 'code': 'invalid_type'}}

Bump. Image input with GPT-4o with Azure AI Foundry Agent was working, but not with GPT-4.1.
